sharpy/knowledges/knowledge.py

```diff
@@ -107,6 +108,16 @@ def set_managers(self, additional_managers: Optional[List[ManagerBase]]):
         if additional_managers:
             self.managers.extend(additional_managers)
 
+    def get_manager(self, manager_type: Type[TManager]) -> TManager:
+        """
+
+        @param manager_type:
+        @return:
+        """
+        for manager in self.managers:
+            if issubclass(type(manager), manager_type):
+                return manager
```

Comment:
I believe list is not the best data structure here.

Iterating the whole list is guaranteed to cause problems for a bot author that eg. uses the `get_manager()` method on every step for every manager. That may not be how `get_manager()` is meant to be used, but it's going to happen.

Reply:
One way to resolve the potential performance problems is to at least warn a bot author about it in the method docstring. Which is included but empty. 😛